### ISSUE TITLE
When installed using MSI, `az upgrade` on 64-bit CLI installs the 32-bit version

### ISSUE THREAD

**Issue by nascarsayan** [Azure/azure-cli#27409]
### Describe the bug

Repro Steps:

OS: Windows
1. Install an old version of 64-bit azure-cli using MSI : `https://azcliprod.blob.core.windows.net/msi/azure-cli-2.51.0-x64.msi`
2. Run `az upgrade`

Expected: Latest 64-bit azure-cli MSI to be downloaded and installed.
Actual: 32-bit azure-cli is downloaded and installed.

Expected: Exit code to indicate success
Actual: Even though installation succeeds, it gives warning that installation failed, and exit code is 1.

### Related command

`az upgrade`

### Errors

NIL

### Issue script & Debug output

```
DEBUG: cli.knack.cli: Command arguments: ['upgrade', '--yes', '--debug']
DEBUG: cli.knack.cli: __init__ debug log:
Cannot enable color.
DEBUG: cli.knack.cli: Event: Cli.PreExecute []
DEBUG: cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments 
at 0x000001E4752E5240>, <function OutputProducer.on_global_arguments at 0x000001E47542FEB0>, <function 
CLIQuery.on_global_arguments at 0x000001E47567D240>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
DEBUG: cli.azure.cli.core: Modules found from index for 'upgrade': ['azure.cli.command_modules.util']
DEBUG: cli.azure.cli.core: Loading command modules:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands
DEBUG: cli.azure.cli.core: util                      0.005         3         7
DEBUG: cli.azure.cli.core: Total (1)                 0.005         3         7
DEBUG: cli.azure.cli.core: Loaded 3 groups, 7 commands.
DEBUG: cli.azure.cli.core: Found a match in the command table.
DEBUG: cli.azure.cli.core: Raw command  : upgrade
DEBUG: cli.azure.cli.core: Command table: upgrade
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function 
AzCliLogging.init_command_file_logging at 0x000001E477D84310>]
DEBUG: cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to 
'C:\Users\Administrator\.azure\commands\2023-09-15.23-05-58.upgrade.4508.log'.
INFO: az_command_data_logger: command args: upgrade --yes --debug
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function 
register_global_subscription_argument.<locals>.add_subscription_parameter at 0x000001E477D86F80>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function 
register_ids_argument.<locals>.add_ids_arguments at 0x000001E477DCCF70>, <function 
register_cache_arguments.<locals>.add_cache_arguments at 0x000001E477DCD090>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument 
at 0x000001E47542FF40>, <function CLIQuery.handle_query_parameter at 0x000001E47567D2D0>, <function 
register_ids_argument.<locals>.parse_ids_arguments at 0x000001E477DCD000>]
WARNING: This command is in preview and under development. Reference and support levels: 
https://aka.ms/CLI_refstatus
DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): raw.githubusercontent.com:443
DEBUG: urllib3.connectionpool: https://raw.githubusercontent.com:443 "GET 
/Azure/azure-cli/main/src/azure-cli/setup.py HTTP/1.1" 200 2406
WARNING: cli.azure.cli.command_modules.util.custom: Your current Azure CLI version is 2.51.0. It will be 
updated to 2.52.0.
WARNING: cli.azure.cli.command_modules.util.custom: Updating Azure CLI with MSI from 
https://aka.ms/installazurecliwindows
DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): aka.ms:443
DEBUG: urllib3.connectionpool: https://aka.ms:443 "GET /installazurecliwindows HTTP/1.1" 301 0
DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): azcliprod.blob.core.windows.net:443
DEBUG: urllib3.connectionpool: https://azcliprod.blob.core.windows.net:443 "GET /msi/azure-cli-2.52.0.msi 
HTTP/1.1" 200 52236288
WARNING: cli.azure.cli.command_modules.util.custom: Downloading MSI to 
C:\Users\ADMINI~1\AppData\Local\Temp\1\tmpnul8hd88\azure-cli-2.52.0.msi
WARNING: cli.azure.cli.command_modules.util.custom: Installing MSI
WARNING: cli.azure.cli.command_modules.util.custom: Succeeded. Deleting 
C:\Users\ADMINI~1\AppData\Local\Temp\1\tmpnul8hd88
WARNING: cli.azure.cli.command_modules.util.custom: CLI upgrade failed or aborted.
DEBUG: cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 
0x000001E477D84550>]
INFO: az_command_data_logger: exit code: 1
INFO: cli.__main__: Command ran in 719.958 seconds (init: 0.485, invoke: 719.473)
INFO: telemetry.main: Begin splitting cli events and extra events, total events: 1
INFO: telemetry.client: Accumulated 0 events. Flush the clients.
INFO: telemetry.main: Finish splitting cli events and extra events, cli events: 1
INFO: telemetry.save: Save telemetry record of length 3224 in cache
INFO: telemetry.main: Begin creating telemetry upload process.
INFO: telemetry.process: Creating upload process: "C:\Program Files\Microsoft SDKs\Azure\CLI2\python.exe 
C:\Program Files\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\telemetry\__init__.pyc 
C:\Users\Administrator\.azure"
INFO: telemetry.process: Return from creating process
INFO: telemetry.main: Finish creating telemetry upload process.
```

### Expected behavior

New 64-bit azure-cli to be installed.
We should download from `https://aka.ms/installazurecliwindowsx64` instead of `https://aka.ms/installazurecliwindows`

Also, exit code does not indicate success. It should indicate success.

### Environment Summary

Pre-upgrade:
```
azure-cli                         2.51.0 *

core                              2.51.0 *
telemetry                          1.1.0

Extensions:
arcappliance                      0.2.33
connectedvmware                    0.2.1
customlocation                     0.1.3
k8s-extension                      1.4.5

Dependencies:
msal                            1.24.0b1
azure-mgmt-resource             23.1.0b2

Python location 'C:\Program Files\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\Administrator\.azure\cliextensions'

Python (Windows) 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:20:36) [MSC v.1929 64 bit (AMD64)]

Legal docs and information: aka.ms/AzureCliLegal


You have 2 update(s) available. Consider updating your CLI installation with 'az upgrade'
```

Post-upgrade:
```
azure-cli                         2.52.0

core                              2.52.0
telemetry                          1.1.0

Extensions:
arcappliance                      0.2.33
connectedvmware                    0.2.1
customlocation                     0.1.3
k8s-extension                      1.4.5

Dependencies:
msal                            1.24.0b1
azure-mgmt-resource             23.1.0b2

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\Administrator\.azure\cliextensions'

Python (Windows) 3.10.10 (tags/v3.10.10:aad5f6a, Feb  7 2023, 17:05:00) [MSC v.1929 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by bebound** [comment]
Thanks for your report, this has been fixed in 2.52
